### ISSUE TITLE
A question about details of the paper

### ISSUE THREAD

**Issue by cleardusk** [yaoyao-liu/meta-transfer-learning#7]
Sorry to raise the questions here, because you authors reply very quickly on github issues : )

In meta-val or meta-test phase, why only update Base-Leaner (the classifier) and keep SS_N fixed? In my opinion, SS_N is also meta-learned, updating SS_N simultaneously seems more straightforward. Are there any other explanations?

Looking forward to your reply!

**Comment by yaoyao-liu** [comment]
During meta-train phase, SS weights are fixed during the base learning process. So, during meta-test phase, SS won't be updated either, which is the same as meta-train. For example, in ProtoNets[1], the feature extractor is updated during meta-learning, but it is fixed in meta-test.

I guess your question is why we don't update SS weights to do task-specific adaptation during base learning. In the experiments, we find that the performance is almost the same if we don't update SS weights during base learning, and it will save lots of GPU memory as we don't need to calculate many second-order gradients. 

Feel free to create issues if you have any question. Thanks for your interest in our work.

[1] Snell, Jake, Kevin Swersky, and Richard Zemel. "Prototypical networks for few-shot learning." Advances in Neural Information Processing Systems. 2017.

**Comment by cleardusk** [comment]
I see. Thanks for your detailed reply!

**Comment by HaofengZhu** [comment]
I have the same confusion about this problem. If you just fix the SS weights when meta-val or  meta-test, it seems to be just using the SS weights to train a more powerful pre-trained net (which may be got in the pre-trained stage if using some good strategy)when mete-training, and it does nothing to help the meta learning since you fix it in meta-test. I'm not sure what I think is true, could you give me some ideas?

Looking forward to your reply!

**Comment by HaofengZhu** [comment]
I have the same confusion about this problem. If you just fix the SS weights during meta-test, it seems to be just using the SS weights to train a more powerful pre-trained net (which may be got in the pre-trained stage if using some good strategy) during mete-train, and it does nothing to help the meta learning since you fix it during meta-test. I'm not sure what I think is true, could you give me some ideas?

Looking forward to your reply!

**Comment by yaoyao-liu** [comment]
Hi @HaofengZhu,

Thanks for your interest in our work.

SS weights are updated during meta-train phase. I think there is no conflict between "improving the performance by meta-learning" and "fixing the weights during meta-test". 

The weights are fixed during meta-test in many few-shot methods, e.g., [ProtoNets](https://arxiv.org/pdf/1703.05175.pdf) and [MathingNets](https://arxiv.org/pdf/1606.04080.pdf).

If you have any further questions, feel free to add more comments.

**Comment by HaofengZhu** [comment]
Hi @yaoyao-liu
In my concept, the SS weights in the meta-train just make the encoder to output better embeddings of the train images, which can make the base learner to better distinguish them. But it seems to do nothing to make the encoder know how to generalize the embedding to all the data(or just for meta-test data), and adjust for the meta-test data. It's like what the image classification based on transfer learning do, but not the meta learning. In the concept of transfer learning, the encoder trained on the training set can also using to other data to get a good embedding.

The  “good strategy” here i think can be using the SS weights on pre-trained (not in meta-train) to train a better net since train a better embedding may also make the training loss lower during pre-trained. The strategy may replace by others which contribute the make a better embedding for training data. 

I think my distinction between meta learning and transfer learning may be too strict, but I just want to exchanges ideas with you.

**Comment by yaoyao-liu** [comment]
Hi @HaofengZhu,

(1) Meta-learning is a kind of transfer learning strategy. In meta-learning, we transfer task-level knowledge instead of sample-level knowledge.

(2) I think you might mix up "task-specific adaption" and "meta-learning". "Meta leaning" includes "task-specific adaptation", "embedding learning". The base-learning updating in [MAML](https://arxiv.org/abs/1703.03400) is a kind of "task-specific adapation" strategy. However, in some meta-learning methods, we do only embedding learning, e.g. [ProtoNets](https://arxiv.org/pdf/1703.05175.pdf). 

(3) To explain why meta-learning performs better than the "good strategy" you mentioned, I recommend you to read [A Closer Look at Few-shot Classification](https://arxiv.org/pdf/1904.04232.pdf). Meta-learning can achieve further improvements on the "good strategy". The detailed experiments and analyses are provided in this paper.

**Comment by HaofengZhu** [comment]
I will read the papers you mentioned carefully. Thanks a lot for your detailed reply!

**Comment by yaoyao-liu** [comment]
> I will read the papers you mentioned carefully. Thanks a lot for your detailed reply!

You are welcome. Feel free to add more comments if you have further questions. Please consider starring our repo and citing our paper if it is helpful to your work.
